src/function/cast/vector_cast_helpers.cpp

```diff
@@ -143,19 +149,116 @@ static bool SplitStringifiedListInternal(const string_t &input, OP &state) {
-	SplitStringOperation state(child_data, child_start, child);
-	return SplitStringifiedListInternal<SplitStringOperation>(input, state);
+bool VectorStringToList::SplitStringList(const string_t &input, string_t *child_data, idx_t &child_start,
+                                         Vector &child) {
+	SplitStringListOperation state(child_data, child_start, child);
+	return SplitStringListInternal<SplitStringListOperation>(input, state);
+}
+
+idx_t VectorStringToList::CountPartsList(const string_t &input) {
+	CountPartOperation state;
+	SplitStringListInternal<CountPartOperation>(input, state);
+	return state.count;
+}
+
+// ------- MAP SPLIT -------
+struct SplitStringMapOperation {
+	SplitStringMapOperation(string_t *child_key_data, string_t *child_val_data, idx_t &child_start, Vector &varchar_key,
+	                        Vector &varchar_val)
+	    : child_key_data(child_key_data), child_val_data(child_val_data), child_start(child_start),
+	      varchar_key(varchar_key), varchar_val(varchar_val) {
+	}
+
+	string_t *child_key_data;
+	string_t *child_val_data;
+	idx_t &child_start;
+	Vector &varchar_key;
+	Vector &varchar_val;
+
+	bool HandleKey(const char *buf, idx_t start_pos, idx_t pos) {
+		if ((pos - start_pos) == 4 && IsNull(buf, start_pos, varchar_key, child_start)) {
+			return false;
+		}
+		child_key_data[child_start] = StringVector::AddString(varchar_key, buf + start_pos, pos - start_pos);
+		return true;
+	}
+
+	void HandleValue(const char *buf, idx_t start_pos, idx_t pos) {
+		if ((pos - start_pos) == 4 && IsNull(buf, start_pos, varchar_val, child_start)) {
+			child_start++;
+			return;
+		}
+		child_val_data[child_start] = StringVector::AddString(varchar_val, buf + start_pos, pos - start_pos);
+		child_start++;
+	}
+};
+
+template <class OP>
+static bool FindKeyOrValueMap(const char *buf, idx_t len, idx_t &pos, OP &state, bool key) {
+	auto start_pos = pos;
+	idx_t lvl = 0;
+	while (pos < len) {
+		if (buf[pos] == '"' || buf[pos] == '\'') {
+			SkipToCloseQuotes(pos, buf, len);
+		} else if (buf[pos] == '{') {
+			SkipToClose(pos, buf, len, lvl, '}');
+		} else if (buf[pos] == '[') {
+			SkipToClose(pos, buf, len, lvl, ']');
+		} else if (key && buf[pos] == '=') {
+			idx_t end_pos = StringTrim(buf, start_pos, pos);
+			return state.HandleKey(buf, start_pos, end_pos); // put string in KEY_child_vector
+		} else if (!key && (buf[pos] == ',' || buf[pos] == '}')) {
+			idx_t end_pos = StringTrim(buf, start_pos, pos);
+			state.HandleValue(buf, start_pos, end_pos); // put string in VALUE_child_vector
+			return true;
+		}
+		pos++;
+	}
+	return false;
+}
+
+template <class OP>
+static bool SplitStringMapInternal(const string_t &input, OP &state) {
+	const char *buf = input.GetDataUnsafe();
+	idx_t len = input.GetSize();
+	idx_t pos = 0;
+
+	SkipWhitespace(buf, pos, len);
+	if (pos == len || buf[pos] != '{') {
+		return false;
+	}
+	SkipWhitespace(buf, ++pos, len);
+	if (buf[pos] == '}') {
```

Comment:
What if pos >= len? Could we add a test for this case (i.e. `{` or `{   `)? Perhaps this will only trigger a segfault if the string is not inlined (e.g. `{                                       `)

Reply:
thnx, I believe it's fixed now and also added a test.